### PR TITLE
Proposal and ADR for release workflow

### PR DESCRIPTION
**Pull request by mbjones**
This PR proposes the release workflow for new proposals and work discussed in issue #30, and the GitFlow-inspired workflow for managing the master, develop, and release branches.  It adds a new CONTRIBUTING.md file describing the process, and an associated ADR.

### CODE REVIEW

**Comment by mbjones** [comment]
After feedback on slack, it was clear we needed a lighter process for simple changes that do not require discussion/decisions via an ADR.  For example, spelling corrections, reformatting, and rewording that doesn't change the intent of a guideline probably don't require an ADR decision. Clarified in sha 9185c7b that maintainers can commit these minor changes directly to `develop`, and other contributors can issue a PR directly against `develop` without using a feature branch.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Contributing to Science on schema.org
+
+The goal of science-on-schema.org is to foster community agreement on guidelines for using schema.org for science-related resource descriptions. Thus, we welcome and encourage community contributions to the work, including proposals for new guidelines, clarifications on the existing guidelines, and proposals for extensions to the schema.org model to accomodate new fields.  Thus, community members are invited to contribute through:
```
Comment (datadavev) [inline CONTRIBUTING.md]: ~~Thus, c~~**C**ommunity members are invited to contribute through:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+- Creating and discussing proposals through [GitHub issues](https://github.com/ESIPFed/science-on-schema.org/issues)
+- Implementing proposed features by editing the guidelines and extension documentation, and editing the Architectural Decision Record for the proposal
```
Comment (datadavev) [inline CONTRIBUTING.md]: Implementing proposed features by editing the guidelines and extension documentation, and editing the Architectural Decision Record **(ADR)** for the proposal

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- Issuing a Pull Request to request review of the proposed changes, and to discuss the feature and iterate on development
```
Comment (datadavev) [inline CONTRIBUTING.md]: Issuing a Pull Request **(PR)** to request review of the proposed changes, and to discuss the feature and iterate on development

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+# Development and Release process
+Because people will review our guidance documents online, it will be helpful to be explicit about versioning and releases of the guidance and other documents.  In particular, in GitHub, users will normally see the `master` branch, which therefore should reflect the current stable release of the documentation (rather thanb confusing people with in-progress proposed changes that are not yet released).
```
Comment (datadavev) [inline CONTRIBUTING.md]: ... (rather than~~b~~ confusing ...

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+Consequently, we will use a [GitFlow](https://nvie.com/posts/a-successful-git-branching-model/)-inspired release model in which the `master` branch always reflects the current stable release, a `develop` branch is used for merging finished proposals being prepared for release, and `feature` branches are used for creating changes to implement specific proposals that are reflected in an Architectural Decision Record (ADR).  For changes that do not require a formal decision via and ADR, such as spelling corrections, grammatical rewording, etc., maintainers can commit changes directly to the `develop` branch, and other contributors can do a pull request directly against the `develop` branch.  The use of feature branches is really focused on managing proposals that need discussion, review, and a decision through an ADR. Maintainers will make judgement calls on whether an ADR is needed, and might convert contributed pull requests to a feature branch if they determine that an ADR is needed.
```
Comment (datadavev) [inline CONTRIBUTING.md]: ... and `feature` branches ~~are used for creating changes to~~ **for** implement**ing** specific ...

... decision via an~~d~~ ADR, ...

File: decisions/30-release-workflow.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# Use Release Process Based on GitFlow
+
+Discussion: https://github.com/ESIPFed/science-on-schema.org/issues/30
+
+## Status ##
+_Proposed_
+
+## Decision ##
+
+We will use a [GitFlow](https://nvie.com/posts/a-successful-git-branching-model/)-inspired release model to track proposed changes, their implementation, and the associated Architectural Decision Record.
```
Comment (datadavev) [inline decisions/30-release-workflow.md]: ... associated Architectural Decision Record **(ADR)**.

File: decisions/30-release-workflow.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+## Context  -- What is the issue that we're seeing that is motivating this decision or change.
+Because people will review our guidance documents online, it will be helpful to be explicit about versioning and releases of the guidance and other documents.  In particular, in GitHub, users will normally see the `master` branch, which therefore should reflect the current stable release of the documentation (rather thanb confusing people with in-progress proposed changes that are not yet released).
```
Comment (datadavev) [inline decisions/30-release-workflow.md]: ... (rather than~~b~~ confusing ...

File: decisions/30-release-workflow.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+Consequently, we will use a [GitFlow](https://nvie.com/posts/a-successful-git-branching-model/)-inspired release model in which the `master` branch always reflects the current stable release, a `develop` branch is used for merging finished proposals being prepared for release, and `feature` branches are used for creating changes to implement specific proposals that are reflected in an Architectural Decision Record.  For changes that do not require a formal decision via and ADR, such as spelling corrections, grammatical rewording, etc., maintainers can commit changes directly to the `develop` branch, and other contributors can do a pull request directly against the `develop` branch.  The use of feature branches is really focused on managing proposals that need discussion, review, and a dec ision through an ADR. Maintainers will make judgement calls on whether an ADR is needed, and might convert contributed pull requests to a feature branch if they determine that an ADR is needed.
```
Comment (datadavev) [inline decisions/30-release-workflow.md]: ... and `feature` branches ~~are used for creating changes to~~ **for** implement**ing** specific ...

... discussion, review, and a ~~dec ision~~**decision** through an ADR. ...

**Comment by mbjones** [comment]
Thanks for all of those, @datadavev

**Pull request merged**